examples/bash_automatic_cd.sh

```diff
@@ -7,14 +7,14 @@
 # To undo the effect of this function, you can type "cd -" to return to the
 # original directory.
 
-function ranger-cd {
-    tempfile="$(mktemp -t tmp.XXXXXX)"
-    ranger --choosedir="$tempfile" "${@:-$(pwd)}"
-    test -f "$tempfile" &&
-    if [ "$(cat -- "$tempfile")" != "$(echo -n `pwd`)" ]; then
-        cd -- "$(cat "$tempfile")"
+ranger-cd() {
+    local temp_file chosen_dir
+    temp_file="$(mktemp)"
+    ranger --choosedir="$temp_file" -- "${@:-$PWD}"
+    if chosen_dir="$(<"$temp_file")" && [[ -n "$chosen_dir" ]]; then
```

Comment:
This doesn't test whether we're already in the dir.

Reply:
Is this required? Does `cd` give any errors? Tested on my machine with `sh`, `bash` and `zsh` and none of them give any errors when `cd`ing into `$PWD`.

Reply:
It's not about errors it's about adding a useless entry to the history. Some people actually use `cd -` to get to the previous dir.

Reply:
Okay, give me a minute.